### PR TITLE
OCPVE-661, OCPBUGS-24415: bump openshift/api version

### PR DESCRIPTION
**Pull request by qJkee**

### CODE REVIEW

**Comment by sadasu** [comment]
@qJkee is this openshift/api version bump still required?

**Review by elmiko — COMMENTED** [review]
the logic here looks correct to me
/lgtm

File: pkg/types/validation/installconfig.go [hunk 1]
```diff
@@ -218,6 +219,17 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
 					"disabling CloudCredential capability available only for baremetal platforms"))
 			}
 		}
+
+		if !enabledCaps.Has(configv1.ClusterVersionCapabilityCloudControllerManager) {
+			if c.None == nil && c.BareMetal == nil && c.External == nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("capabilities"), c.Capabilities,
+					"disabling CloudControllerManager available only for non cloud platforms or External type with cloudControllerManager value none"))
```
Comment (patrickdillon) [inline pkg/types/validation/installconfig.go:226]:
```suggestion
					"disabling CloudControllerManager is only supported on the Baremetal, None, or External  platforms"
```

File: pkg/types/validation/installconfig.go [hunk 2]
```diff
@@ -218,6 +219,17 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
...
+			}
+			if c.External != nil && c.External.CloudControllerManager == external.CloudControllerManagerTypeExternal {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("capabilities"), c.Capabilities,
+					"disabling CloudControllerManager available only for cloudControllerManager value none"))
```
Comment (patrickdillon) [inline pkg/types/validation/installconfig.go:230]:
```suggestion
					"disabling CloudControllerManager supported only with cloudControllerManager value none"))
```

**Review by patrickdillon — COMMENTED** [review]
/approve

Thanks for taking care of this. A couple of suggestions for reworking the error messages.

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Pull request merged**